climada_petals/entity/exposures/openstreetmap/osm_dataloader.py

```diff
@@ -0,0 +1,945 @@
+        """
+        
+        if not Path(save_path).is_file():
+            LOGGER.info(f'Downloading file as {save_path}')
+            urllib.request.urlretrieve(self.planet_url, save_path)
+        else:
+            LOGGER.info(f'file already exists as {save_path}')
+
+    def _osmosis_extract(self, shape, path_planet, path_extract):
+        """
+        Runs the command line tool osmosis to cut out all map info within 
+        shape, from the osm planet file, unless file already exists.
+        
+        Parameters
+        -----------
+        shape : list or str
+            bounding box [xmin, ymin, xmax, ymax] or file path to a .poly file
+        path_planet : str or pathlib.Path
+            file path to planet.osm.pbf
+        path_extract : str or pathlib.Path
+            file path (incl. name & ending) under which extract will be stored
+            
+        Returns
+        -------
+        None
+            Extracts and stores a new file as indicated under path_extract
+        """
+
+        if not Path(path_extract).is_file():
+            
+            LOGGER.info('File doesn`t yet exist. Assembling osmosis command.')
+            if (isinstance(shape, list) or isinstance(shape, tuple)):
+                cmd = ['osmosis', '--read-pbf', 'file='+str(path_planet), 
+                       '--bounding-box', f'top={shape[3]}', f'left={shape[0]}',
+                       f'bottom={shape[1]}', f'right={shape[2]}', 
+                       '--write-pbf', 'file='+str(path_extract)]
+            elif isinstance(shape, str):
+                cmd = ['osmosis', '--read-pbf', 'file='+str(path_planet), 
+                       '--bounding-polygon', 'file='+shape, '--write-pbf',
+                       'file='+str(path_extract)]
+
+            LOGGER.info('''Extracting from the osm planet file... 
+                        This will take a while''')
+
+            return subprocess.run(cmd, stdout=subprocess.PIPE, 
+                                  universal_newlines=True)
+        else:
+            LOGGER.info("Extracted file already exists!")
+            return None
+
+    def get_data_planetextract(self, shape, path_extract,
+                               path_planet=Path(DATA_DIR, 'planet-latest.osm.pbf')):
+        """
+        get OSM raw data from a custom shape / bounding-box, which is extracted
+        from the entire OSM planet file. Accepts bbox lists or .poly files for
+        non-rectangular shapes.
+
+        Parameters
+        ----------
+        shape : list or str
+            bounding box [xmin, ymin, xmax, ymax] or file path to a .poly file
+        path_extract : str or pathlib.Path
+            file path (incl. name & ending) under which extract will be stored
+        path_planet : str or pathlib.Path
+            file path to planet-latest.osm.pbf. Will download & store it as 
+            indicated, if doesn`t yet exist.
+            Default is DATA_DIR/planet-latest.osm.pbf
+
+
+        Note
```

Comment:
Could you add a note here on how to install the osmosis command line tool? (And possibly check for its existence and throw an error if it's not found?)

Reply:
also added it to the tutorial and several other parts now: 3906d0c0

Reply:
nice